packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,672 @@
+  <ActionList.Header title="Recent Searches" icon={HistoryIcon} />
+  <ActionList.SectionHeading title="Hello" />
+  <ActionList.Item title="Item Value" desciption="Item Description" variant="negative">
+    <ActionList.Leading>
+      <StarIcon />
+    </ActionList.Leading>
+    <ActionList.Trailing>
+      <Text>⌘ + Space</Text>
+    </ActionList.Trailing>
+  </ActionList.Item>
+  <ActionList.Divider />
+  <ActionList.Footer title="Footer Title" description="Footer Description">
+    <ActionList.Leading>
+      <DocIcon />
+    </ActionList.Leading>
+    <ActionList.Trailing>
+      <Button variant="secondary" onClick={}>
+        Secondary Button
+      </Button>
+      <Button onClick={}>Primary Button</Button>
+    </ActionList.Trailing>
+  </ActionList.Footer>
+</ActionList>
+```
+
+</details>
+
+### Custom Implementation instead of Library
+
+> **Note**
+>
+> Didn't get too much into implementation details during API decisions. Would recommend evaluating this once again while implementing the Dropdown.
+
+TLDR; Felt like our Dropdown was a bit more complicated than other dropdowns we've seen so we might have to fight against existing opinions from libraries if we go with libraries.
+
+Some libraries we evaluated
+
+- [downshift-js](https://github.com/downshift-js/downshift)
+
+  If we decide to go with library, this will definitely be the first thing we can explore.
+
+  - 😄 Covers the accessibility and gives controls to consumers for styling
+  - 🙁 Their API seemed a bit verbose and difficult to implement at first so unsure if it's worth adding library. (We should definitely refer to their Markup for accessibility practices though)
+  - 😐 Around 20kb (~10kb for Select and ~10kb for MultiSelect)
+
+- [@radix-ui/react-dropdown-menu](https://www.radix-ui.com/docs/primitives/components/dropdown-menu)
+  - 😄 Covers the accessibility gives basic styling
+  - 🙁 The items are also `Dropdown` compound components so we will have to tie our `ActionList.Item` to `DropdownMenu.Item`. Which means `ActionList` can't be used outside of `Dropdown`?
+  - 🙁 Opinionated so we might face more of such ^^ blockers as we discover more usecases.
+
+<br/><br/>
+
+## Accessibility
+
+- For single select, we should have `role=menu` on the container and `role=menuitem` on list items
+- For multiple select,
+
+  - we should have `role=listbox` and `aria-multiselectable=true` on the container (because `role=menu` is expected to close after one item is selected)
+  - Input search/filter should be outside of this container
+    Refer: https://primer.style/react/storybook/?path=/story/components-selectpanel--multi-select-story
+
+- Dropdown is supposed to open when you focus on select and press `ENTER`, `SPACE` or any of the arrow keys. Arrow Up and Down can be pressed to move to next and previous items in the action list. Should close on pressing `ESC`.
+
+<br/><br/>
+
+## Open Questions
+
+### Dev
+
+- Vote: `variant` or `type` attribute on `Dropdown`?
```

Comment:
what if we have `selectionType: single | multiple` or `itemSelection: single | multiple`

Reply:
I think I'll prefer `type` in that case instead of introducing new prop. `type` also conveys the similar thing imo 🤔

Reply:
changed it to selectionType since that is what we use in design as well